src/nft/components/collection/Card.tsx

```diff
@@ -0,0 +1,383 @@
+import clsx from 'clsx'
+import Column from 'components/Column'
+import { Box } from 'nft/components/Box'
+import { Row } from 'nft/components/Flex'
+import { MinusIconLarge, PlusIconLarge } from 'nft/components/icons'
+import { body, subheadSmall } from 'nft/css/common.css'
+import { themeVars, vars } from 'nft/css/sprinkles.css'
+import { useIsMobile } from 'nft/hooks'
+import { GenieAsset } from 'nft/types'
+import {
+  createContext,
+  MouseEvent,
+  ReactNode,
+  useContext,
+  useLayoutEffect,
+  useMemo,
+  useReducer,
+  useRef,
+  useState,
+} from 'react'
+
+import * as styles from './Card.css'
+
+/* -------- ASSET CONTEXT -------- */
+export interface CardContextProps {
+  asset: GenieAsset
+  hovered: boolean
+  selected: boolean
+  href: string
+  setHref: (href: string) => void
+}
+
+const CardContext = createContext<CardContextProps>({
+  asset: {} as GenieAsset,
+  hovered: false,
+  selected: false,
+  href: '',
+  setHref: () => undefined,
+})
+
+const useCardContext = () => {
+  return useContext(CardContext)
+}
+
+const baseHref = (asset: GenieAsset) => `/#/nft/asset/${asset.address}/${asset.tokenId}?origin=collection`
+
+/* -------- ASSET CARD -------- */
+interface CardProps {
+  asset: GenieAsset
+  children: ReactNode
+}
+
+export const Card = ({ asset, children }: CardProps) => {
+  const [hovered, toggleHovered] = useReducer((s) => !s, false)
+  const [href, setHref] = useState(baseHref(asset))
+
+  const providerValue = useMemo(
+    () => ({
+      asset,
+      selected: false,
+      hovered,
+      toggleHovered,
+      href,
+      setHref,
+    }),
+    [asset, hovered, href]
+  )
+
+  const assetRef = useRef<HTMLDivElement>(null)
+
+  useLayoutEffect(() => {
```

Comment:
is this necessary if we have the onMouseEnter and leave stuff below?

Reply:
yes there are states where onMouseEnter doesn't trigger - for instance if the card is loaded underneath the user's cursor